### ISSUE TITLE
Add support for the general sibling combinator

### ISSUE THREAD

**Issue by ernsto** [MyIntervals/emogrifier#301]
Hello, 

i have an problem with the parsing of my CSS code. Some CSS rules cause warnings because they can not be converted into a valid xpath query. 

I debugged the Emogrifier class and found out following: 
The CSS rules are translated to xpath queries with the `translateCsstoXpath` method in the line 353 of the `Emogrifier` class. 

```
$nodesMatchingCssSelectors = $xPath->query($this->translateCssToXpath($cssRule['selector']));
```

Here are some examples of CSS rules which are translated into invalid xpath queries: 

```
CSS: .pager-no-toolbar ~ .pager-no-toolbar
Xpath: //*[contains(concat(" ",@class," "),concat(" ","pager-no-toolbar"," "))]//~//*[contains(concat(" ",@class," "),concat(" ","pager-no-toolbar"," "))]
```

```
CSS: .col-left .block .actions .button ~ a
Xpath: //*[contains(concat(" ",@class," "),concat(" ","col-left"," "))]//*[contains(concat(" ",@class," "),concat(" ","block"," "))]//*[contains(concat(" ",@class," "),concat(" ","actions"," "))]//*[contains(concat(" ",@class," "),concat(" ","button"," "))]//~//a
```

```
CSS: .col-right .block .actions .button ~ a
Xpath: //*[contains(concat(" ",@class," "),concat(" ","col-right"," "))]//*[contains(concat(" ",@class," "),concat(" ","block"," "))]//*[contains(concat(" ",@class," "),concat(" ","actions"," "))]//*[contains(concat(" ",@class," "),concat(" ","button"," "))]//~//a
```

```
CSS: .block-layered-nav .block-content dl dd ol>li .ui-slider.ui-slider-pips [class*=i-slider-pip-selected]
Xpath: //*[contains(concat(" ",@class," "),concat(" ","block-layered-nav"," "))]//*[contains(concat(" ",@class," "),concat(" ","block-content"," "))]//dl//dd//ol>li//*[contains(concat(" ",@class," "),concat(" ","ui-slider"," "))][contains(concat(" ",@class," "),concat(" ","ui-slider-pips"," "))]//[class*=i-slider-pip-selected]
```

Following Warning occurs: 

```
Warning: DOMXPath::query(): Invalid expression in <path to php class>/Emogrifier.php on line 353
```

I am using the latest version of the class (87d9d61). 

Is it possible to create a valid xpath query for the CSS rules?

Thank you for your help.

**Comment by oliverklee** [comment]
The general sibling combinator currently is not supported yet.  Would you be willing to provide a pull request that provides this feature?

**Comment by oliverklee** [comment]
This was fixed in #301.
